src/ui/server/middleware/insurance/application-status/index.ts

```diff
@@ -0,0 +1,33 @@
+import { INSURANCE_ROUTES } from '../../../constants/routes/insurance';
+import { APPLICATION } from '../../../constants';
+import { Next, Request, Response } from '../../../../types';
+
+const { APPLICATION_SUBMITTED, NO_ACCESS_APPLICATION_SUBMITTED } = INSURANCE_ROUTES;
+
+export const applicationStatusMiddleware = async (req: Request, res: Response, next: Next) => {
+  const { baseUrl: url } = req;
+
+  const isApplicationSubmittedRoute = (route: string) => route.includes(APPLICATION_SUBMITTED);
+
+  // do not run any checks if the requested route is: APPLICATION_SUBMITTED
+  if (isApplicationSubmittedRoute(url)) {
+    return next();
+  }
+
+  /**
+   * If the status is submitted,
+   * dont allow the user to view/access the application and redirect to NO_ACCESS_APPLICATION_SUBMITTED.
+   * Otherwise, redirect to the application.
+   */
+  if (res.locals.application) {
```

Comment:
Do we need optional chaining here?

Reply:
The middleware run before this middleware will catch this, so before this is run, if there is no application in `res.locals` it'll redirect to `NO_ACCESS_TO_APPLICATION`